geoengine/layers.py

```diff
@@ -0,0 +1,660 @@
+
+    def _repr_html_(self) -> str | None:
+        '''HTML representation for Jupyter notebooks'''
+
+        buf = StringIO()
+
+        buf.write("<table>")
+        buf.write('<thead><tr><th colspan="2">Layer Collection</th></tr></thead>')
+        buf.write("<tbody>")
+        buf.write(f"<tr><th>name</th><td>{self.name}</td></tr>")
+        buf.write(f"<tr><th>description</th><td>{self.description}</td></tr>")
+        buf.write(f"<tr><th>id</th><td>{self.collection_id}</td></tr>")
+        buf.write(f"<tr><th>provider id</th><td>{self.provider_id}</td></tr>")
+
+        num_items = len(self.items)
+        for (i, item) in enumerate(self.items):
+            buf.write('<tr>')
+            if i == 0:
+                buf.write(f'<th rowspan="{num_items}">items</th>')
+            buf.write(f"<td>{item.html_str()}</td>")
+            buf.write('</tr>')
+
+        buf.write("</tbody>")
+        buf.write("</table>")
+
+        return buf.getvalue()
+
+    def remove(self, timeout: int = 60) -> None:
+        '''Remove the layer collection itself'''
+
+        if self.provider_id != LAYER_DB_PROVIDER_ID:
+            raise ModificationNotOnLayerDbException('Layer collection is not stored in the layer database')
+
+        _delete_layer_collection(self.collection_id, timeout)
+
+    def remove_item(self, index: int, timeout: int = 60):
+        '''Remove a layer or collection from this collection'''
+
+        if index < 0 or index >= len(self.items):
+            raise IndexError(f'index {index} out of range')
+
+        item = self.items[index]
+
+        if self.provider_id != LAYER_DB_PROVIDER_ID:
+            raise ModificationNotOnLayerDbException('Layer collection is not stored in the layer database')
+
+        if item.type == LayerCollectionListingType.LAYER:
+            _delete_layer_from_collection(
+                self.collection_id,
+                cast(LayerId, item.listing_id),
+                timeout,
+            )
+        elif item.type == LayerCollectionListingType.COLLECTION:
+            _delete_layer_collection_from_collection(
+                self.collection_id,
+                cast(LayerCollectionId, item.listing_id),
+                timeout,
+            )
+
+        self.items.pop(index)
+
+    def add_layer(self,
+                  name: str,
+                  description: str,
+                  workflow: Dict[str, Any],  # TODO: improve type
+                  symbology: Optional[Dict[str, Any]],  # TODO: improve type
+                  timeout: int = 60) -> LayerId:
+        '''Add a layer to this collection'''
+        # pylint: disable=too-many-arguments
+
+        if self.provider_id != LAYER_DB_PROVIDER_ID:
+            raise ModificationNotOnLayerDbException('Layer collection is not stored in the layer database')
+
+        layer_id = _add_layer_to_collection(name, description, workflow, symbology, self.collection_id, timeout)
+
+        self.items.append(LayerCollectionListing(
```

Comment:
is that a good idea? maybe due to pagination and sorting it wouldn't directly be visible? Maybe reload the items? Maybe differentiate between a Collection and a CollectionPage?

Reply:
For now, we load everything until we have more experience with the API:
[c6d8879](https://github.com/geo-engine/geoengine-python/pull/83/commits/c6d8879dbbde1ba5110530339e524e9f1e3a9ef2)